app/services/metrics/base_metric_processor.rb

```diff
@@ -0,0 +1,84 @@
+module Metrics
+  ##
+  # Base class for classes that do the processing of a Metric from a
+  # collection of events.
+  class BaseMetricProcessor < BaseService
+    def initialize(events:, time_interval:)
+      @events = events
+      @time_interval = time_interval
+    end
+
+    ##
+    # Processes the given events to generate the metrics.
+    # The given @events are expected to include all the events during the full
+    # time interval ( [@starting_at, @starting_at + @time_span) ).
+    def call
+      process_events(events: @events, time_interval: @time_interval)
```

Comment:
you don't need to pass those params, they are sent in the initialization and can be used as instance variables

Reply:
merging this comment with the one above, I'd just remove the call method and make `process_events` public (without params)